### PR TITLE
chore: Switch to kind ClusterKeycloak in case if add-ons approach use

### PR DESCRIPTION
**Pull request by alezander86**
Description
This pull request introduces conditional logic and configuration updates for the Keycloak deployment templates within the deploy-templates directory. The aim is to refine the deployment process based on the integration of external secrets and the presence of an OpenID Connect (OIDC) extension. The updates ensure that only the necessary Keycloak resources are created and managed, preventing redundant configurations when the OIDC extension is already in place.

Key Changes:
Added conditions to check the state of .Values.externalSecrets.enabled and .Values.sso.addonsExtensionsOIDC to optimize secret management.
Revised references to keycloakRef to align with the new conditional logic, ensuring that Keycloak resources are referenced correctly.
Expanded the values.yaml to include configurations for the addonsExtensionsOIDC flag, which dictates the creation of additional Keycloak resources based on the presence of an OIDC extension.
These modifications will provide greater flexibility in deploying Keycloak within environments that use external secrets management and have the OIDC extension installed.

Type of change
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
 
How Has This Been Tested?
 - [x] Tests are included to verify the logic of the conditional statements.
 - [x] Deployment has been tested in an environment where the OIDC extension is present to ensure no additional - [ ] - 
 - [x]  Keycloak resources are created unnecessarily.

Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Pull Request contain one commit. I squash my commits

### CODE REVIEW

File: deploy-templates/templates/keycloak/realm.yaml [hunk 1]
```diff
@@ -6,7 +6,14 @@ metadata:
   labels:
     {{- include "edp-install.labels" . | nindent 4 }}
 spec:
-  keycloakOwner: main
+  keycloakRef:
+  {{- if .Values.sso.addonsExtensionsOIDC }}
```
[the remark this answers is not in the document]
Comment (alezander86) [reply deploy-templates/templates/keycloak/realm.yaml]: done

File: deploy-templates/templates/keycloak/realmrolebatch.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.sso.enabled -}}
+{{- if and .Values.sso.enabled (not .Values.sso.addonsExtensionsOIDC) -}}
```
Comment (NikolayMarusenko) [inline deploy-templates/templates/keycloak/realmrolebatch.yaml:1]: do we need this? the feeling that we are not creating this resource in addons?

Comment (alezander86) [reply deploy-templates/templates/keycloak/realmrolebatch.yaml]: done

File: deploy-templates/templates/keycloak/realmrolebatch.yaml
Comment (Rolika4) [inline deploy-templates/templates/keycloak/realmrolebatch.yaml:1]: {{- if .Values.sso.enabled -}}

**Review by Rolika4 — DISMISSED** [review]

File: deploy-templates/values.yaml [hunk 3]
```diff
@@ -452,6 +452,15 @@ edp-tekton:
 sso:
   # -- Install OAuth2-proxy and Keycloak CRs as a part of EDP deployment.
   enabled: false
+  # Enable if using the addons approach (see https://github.com/epam/edp-cluster-add-ons)
+  # for EDP installation and the extension-oidc is already installed.
+  # This prevents the creation of an additional Keycloak resource and secret.
```
Comment (NikolayMarusenko) [inline deploy-templates/values.yaml:457]: keycloakOperatorResources:
   # Create kind: Keycloak/ClusterKeycloak as a part of chart installation
  createKeycloakCR: true
  # Can be Keycloak or ClusterKeycloak. 
  kind: Keycloak
  # Name of kind: Keycloak/ClusterKeycloak CR;
  name: main

**Review by NikolayMarusenko — APPROVED** [review]

**Pull request merged**